test/credential-test.js

```diff
@@ -156,6 +156,59 @@ test('verify with empty password', function(t) {
 
 });
 
+test('constantEquals', function (t) {
+  function timed_compare(a, b) {
+    var start = process.hrtime();
+    pw.constantEquals(a, b);
+    return process.hrtime(start)[1];
+  }
+  var i,
+      iterations = 25000,
+      equal_results = 0,
+      inequal_results = 0,
+      difflen_results = 0;
+  // Ensure it works
+  t.ok(pw.constantEquals("abc", "abc"), 'equality')
+  t.ok(pw.constantEquals("", ""), 'equal-empty')
+  t.ok(!pw.constantEquals("a", ""), 'inequal 1-char')
+  t.ok(pw.constantEquals("a", "a"), 'equal 1-char')
+  t.ok(!pw.constantEquals("ab", "ac"), 'inequal 2-char')
+  t.ok(pw.constantEquals("ab", "ab"), 'equal 2-char')
+  t.ok(!pw.constantEquals("abc", "abC"), 'inequality - difference')
+  t.ok(!pw.constantEquals("abc", "abcD"), 'inequality - addition')
+  t.ok(!pw.constantEquals("abc", "ab"), 'inequality - missing')
+
+  // Ensure timing is sane
+  // Differing lengths
+  for (i = 0; i < iterations; i++) {
+    difflen_results += timed_compare("abcd", "abcdefghijklmnopqrstuvwxyz");
+  }
+
+  for (i = 0; i < iterations; i++) {
+    equal_results   += timed_compare("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyz");
+  }
+
+  for (i = 0; i < iterations; i++) {
+    inequal_results += timed_compare("abcdefghijklmnopqrstuvwxyz", "ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+  }
+
+  // This is a point of some statistical importance. A tolerance of 
```

Comment:
This is where I'm out of my depth, and I'd really like a statistical timing expert to review our tests. Do you know somebody we could ask?